force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,64 @@
+@isTest
+public with sharing class LeadTriggerHandlerTest {
+
+    @isTest
+    public static void testLeadTitleNormalization() {
+        List<Lead> leads = new List<Lead>();
+        Lead lead = new Lead(FirstName = 'Test', LastName = 'Test', Title = 'vp', Company = 'Test');
+        leads.add(lead);
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Title = 'mgr', Company = 'Test'));
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Title = 'exec', Company = 'Test'));
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Title = 'assistant', Company = 'Test'));
+
+        
+        
+        Test.startTest();
+        LeadTriggerHandler.handleTitleNormalization(leads);
+        Test.stopTest();
+        Lead updatedLead = leads[0];
+        Assert.areEqual('Vice President', updatedLead.Title, 'Title should be Vice President');
+        Assert.areEqual('Manager', leads[1].Title, 'Title should be Manager');
+        Assert.areEqual('Executive', leads[2].Title, 'Title should be Executive');
+        Assert.areEqual('Assistant', leads[3].Title, 'Title should be Assistant');
+    }
+
+    @isTest
+    public static void testHandleAutoLeadScoring() {
+     
+        List<Lead> leads = new List<Lead>();
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Company = 'Test', Title = 'vp', LeadSource = 'Website', Industry = 'Technology', Phone = '4055141430', Email ='j.matt.harris@gmail.com'));
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Company = 'Test', Title = 'vp', LeadSource = 'Website', Industry = 'Technology', Phone = '4055141430'));
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Company = 'Test', Title = 'vp', LeadSource = 'Website', Industry = 'Technology'));
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Company = 'Test', Title = 'vp', LeadSource = 'Website'));
+        leads.add(new Lead(FirstName = 'Test', LastName = 'Test', Company = 'Test', Title = 'vp'));
```

Comment:
As an advanced tip sometimes I'll put the value I am looking for in a field on the record for example. Lastname = 18 so that I can dynamically check if the scores calculated correctly.